lib/util/util.CombiningMessageProvider.js

```diff
@@ -0,0 +1,53 @@
+this.util = this.util || {};
+
+util.CombiningMessageProvider = ( function() {
+	'use strict';
+
+	/**
+	 * Providing messages using two message providers
+	 * @class util.CombiningMessageProvider
+	 * @licence GNU GPL v2+
+	 * @author H. Snater < mediawiki@snater.com >
+	 * @author Adrian Heine < adrian.heine@wikimedia.de >
+	 *
+	 * @constructor
+	 *
+	 * @param {util.MessageProvider} a
+	 * @param {util.MessageProvider} b
+	 */
+	function CombiningMessageProvider( a, b ) {
```

Comment:
1. I've used `Combining` for the entity stores as well. I'm open for another name, though :)
1. I find it easier not having to implement iteration, and I have no use-case for more than two. An array would work for me, too, though.